### ISSUE TITLE
jshint should warn when using undeclared/undefined default values (in function parameters)

### ISSUE THREAD

**Issue by julienw** [jshint/jshint#2327]
Currently jshint does not seem to warn for these pieces of code:

```javascript
// "global" is not defined
function test1(param = global) {
}

// "param" is declared, but always has the value undefined so it's likely not what the author intended.
function test2(param = +param) {
}
```

**Comment by rwaldron** [comment]
This is a good one.

**Comment by lukeapage** [comment]
related: #2422

**Comment by lukeapage** [comment]
this is fixed in 2.8.0 if you have undef: true on..

```
/* jshint undef: true */
function test1(param = global) {
}
```
